### DISCUSSION TITLE
SONOFF TRVZB – F5 error after firmware update

### DISCUSSION THREAD

**Discussion by nelhop** [Koenkk/zigbee2mqtt#24518] [Devices]
Unfortunately my home assistant was rebooted while a firmware update of SONOFF TRVZB Thermostat (from 4357 to 4609) was running.

Now the Thermostat seems to be broken and is not responding.
It switches between "CA" and "20°" several times before showing "F5". F5 means that no valve is detected, but it is installed the same way it was when it worked correctly.

A factory reset of the Thermostat (by removing the batteries and inserting them again while pushing the button) does not change anything.

Is there any way to reinstall or reset the firmware?

**Comment by niall-123** [comment]
Same for me with 4 TRVZBs, upgraded to 1.2.1 and all displaying F5 now. No reboot of home assistant during the process and the update appears to have completed successfully. Any joy in resolving?

**Reply by nelhop** [reply]
Nope the problem still persists. Good to know that it is not due to the HA reboot though.

**Comment by WoodenDuke** [comment]
Yesterday I updated one of my 12 TRVZBs to version 1.2.1. Took almost an hour to complete. Afterwards I had to remove the batteries and restart the thermostat, but now it works perfectly. So it doesn't seem to be a general problem.

**Comment by Aabanken** [comment]
Same problem for me with 2 TRVZBs, upgraded to 1.2.1 and all displaying F5 now.

**Reply by WoodenDuke** [reply]
That's interesting - why didn't that happen to my TRVZB? Any ideas?
At the moment I'm pretty unsure, If I should update the other ones...

**Reply by niall-123** [reply]
I have 7 TRVZBs in total. 3 bought earlier in the year and 4 this month. All updated to 1.2.1 but only the 4 bought this month are showing the F5 issue. The 3 earlier ones are working fine on 1.2.1. Can't recall what the original firmware was on any of the TRVZBs.

**Reply by WoodenDuke** [reply]
I have 12 of them, all purchased in March of this year. They all previously had firmware version 4357 (1.1.5).

Now I have updated **all** of them, but I only rebooted two of them after the firmware update by removing the batteries (they just stopped responding after the update - waited an hour and then rebooted).

Now **all thermostats are on version 1.2.1 (4609) and work perfectly**. None of them showed any error in the entire process.

**Reply by srett** [reply]
I have two of them and updated one so far, successfully.

Both of them had 1.1.1 or whatever the older firmware from 2023 was. I already updated them to 1.1.5 a while ago, and one of them for some reason was still reported as having 1.1.1 on it after the update, but at the same time, there was no update-button next to it, so this might have been an issue with z2m entirely. Now it's updated to 1.2.1 and reports as such, the other one I didn't do yet.

**Comment by dragosj** [comment]
Updated 11 TRVZB to 1.2.1 using Z2M OTA and all went fine with no need to reset or reboot. All were bought last year. Maybe it is a different hardware in new ones?

**Comment by SirUli** [comment]
Had the same issue. What I found was that it is resolved when I push the TRVZB towards the valve while it adjusts. No idea why but that helped to solve the error. The TRVZBs were installed for about half a year, purchased at rhe end of 2023 and only showed the error after the upgrade.

**Reply by niall-123** [reply]
Thanks for sharing this, it worked for me too! I had to remove all fully, put back on the radiators and then push gently while they adjusted. I'm using the adaptors so not sure if that had anything to do with it but all working on 1.2.1 now.

**Reply by SirUli** [reply]
Unfortunately no permanent solution as the decalcification process re-calibrates and then it is again F5...

**Comment by IndiGP** [comment]
If I had read this before... It' annoying.. is there a possibility to revoke the update? I think it's running with Open valve then, right?

**Comment by DavidGeorge528** [comment]
I updated 8 TRVs. All from v1.1.5 to v1.2.1. 2 were bought when the TRVs were first released and the other 6 were bought last month. 

5 of them now work fine and 3 are stuck showing F5. One of my original ones and 2 from the new batch. So there isn't any consistency with how old the device is, or the starting firmware. All devices showing F5 weren't touched during the update. And I've tried the suggestions to get the error to go away but nothings worked yet.

**Reply by srett** [reply]
Are you using an adapter?

**Reply by DavidGeorge528** [reply]
No, the TRV is screwed directly onto the radiator valve

**Reply by srett** [reply]
Hm, same here. Just finished updating my other one and no issues. Since you got 8 of them, can you find any pattern or similarity among the working and non-working ones? It looks more and more like this indeed random, but it's still odd that a few people could fix the problem by fiddling with the TRV while powering it up. Maybe the "no valve detected" code got tweaked and is a bit too eager now?

**Reply by WoodenDuke** [reply]
My 12 TRVZBs are also mounted directly without an adapter. As mentioned above, all are now error-free on 1.2.1, only some had to be restarted after the update. So no hint either.

**Reply by DavidGeorge528** [reply]
I did some testing.

Swapping my TRV to a valve which doesn't give an error on a different TRVZB removed the error. It seems Sonoff altered the lowest point of the TRVZB so it doesn't bottom out on some of my valves.

I suspected this when others reported fixing the issue by pushing down on it as it calibrates.

I fixed the issue on all 3 of by TRVZBs showing F5 by sticking a small metal disk (1-2mm thick) to the plunger and reinstalling. The error went away on all

**Reply by srett** [reply]
> I fixed the issue on all 3 of by TRVZBs showing F5 by sticking a small metal disk (1-2mm thick) to the plunger and reinstalling.

Hah, I used a two cent coin like that ages ago when a plain old "dumb" TRV didn't fully close the valve. :-)

**Reply by gianlucavolt** [reply]
Same for me, the coin worked it out straightforward. Strange is that only one over 5 valves had this issue after the update...

**Reply by sdechevigne** [reply]
> I did some testing.
> 
> Swapping my TRV to a valve which doesn't give an error on a different TRVZB removed the error. It seems Sonoff altered the lowest point of the TRVZB so it doesn't bottom out on some of my valves.
> 
> I suspected this when others reported fixing the issue by pushing down on it as it calibrates.
> 
> I fixed the issue on all 3 of by TRVZBs showing F5 by sticking a small metal disk (1-2mm thick) to the plunger and reinstalling. The error went away on all

I had the same problem but I leaned the valve against a wall with a metal object on the black part. And it's fine.

**Comment by superfrobo** [comment]
Same problem. On one heater the updated valve showed F5. changed with a valve with previous firmware and this worked. Updated this new valve and it does not work again. Changed with another old valve and this time I’m not going to update anything anymore. This Chinese are really not able to write the firmware at all.

**Reply by srett** [reply]
Instead of resorting to racism, maybe take the time to read the suggested solutions here and confirm you tried them all.

**Reply by SirUli** [reply]
@superfrobo how did you downgrade? Thanks in advance!

**Reply by superfrobo** [reply]
> @superfrobo how did you downgrade? Thanks in advance!

No way to downgrade. I bought another valve with old firmware on it …

**Comment by superfrobo** [comment]
The best solution would be test the firmware with european standards before selling devices to european countries. But Chinese dont' like to comply. We buy any kind of rubbish from this country indeed. The above solutions are ridicolous... putting metal rings, pushing the valve while pressing the button. I payed hundred of Euro for these devices. If you would read the answers the gave me to my suggestion to improve their software you would creepy.

**Reply by a4refillpad** [reply]
What's with these sinophobic rants? These issues are common with trvs if you've had any experience. I had to do similar things with my energenie etrv (not Chinese) too which I'm upgrading from.

**Comment by jonathh21** [comment]
Just here to throw in a 'me too' just after I have ordered a stack more.. my current one breaks most firmware update.

**Reply by jonathh21** [reply]
> Just here to throw in a 'me too' just after I have ordered a stack more.. my current one breaks most firmware update.

Further to my quick 'me too' and after having changed batteries, left empty, removed and readded nothing seemed to work.

I reached out to sonoff who were very responsive (which was reassuring as I have just ordered 5 more of these!)

They confirmed it is a calibration error and apparently they know there is an issue post update.

Asked me to ensure my adapter was as tight as possible. So I tightened up and then got some monkey grips and tightened up a small amount further.. and it worked!

My trv is now functional again.

Not sure what benefits the update has given me to be worth the extra faff/risk of damaging the fixings or value tho.

**Reply by SirUli** [reply]
I also opened up a ticket to sonoff now as this is a annoying behaviour. It worked before 1.2.1 and still works on those were i stopped the upgrade. But all upgraded ones are useless and are currently replaced by the "old & dumb" thermostats to actually get my heating back to work. The tightening of the adapter is in my case impossible as the Danfoss RAVL adapter simply starts slipping when you go too far.

**Reply by trolle4** [reply]
I did open one as well now. The more, the merrier ;) 

I just updated one thermostat and got it working by gluing a small piece of plastic onto the cylinder to make it reach the pin on the valve before displaying F5. Not a long term solution, but it does indeed seem like the reach of the cylinder has been decreased by the firmware update. I described this in the ticket, so I expect the firmware fix should be pretty straight forward ;)

**Reply by jonathh21** [reply]
> > Just here to throw in a 'me too' just after I have ordered a stack more.. my current one breaks most firmware update.
> 
> Further to my quick 'me too' and after having changed batteries, left empty, removed and readded nothing seemed to work.
> 
> I reached out to sonoff who were very responsive (which was reassuring as I have just ordered 5 more of these!)
> 
> They confirmed it is a calibration error and apparently they know there is an issue post update.
> 
> Asked me to ensure my adapter was as tight as possible. So I tightened up and then got some monkey grips and tightened up a small amount further.. and it worked!
> 
> My trv is now functional again.
> 
> Not sure what benefits the update has given me to be worth the extra faff/risk of damaging the fixings or value tho.

Not sure if time has loosened it, but today the f5 error returned.

I hope we get a fix soon as I have a few active sonof trvs that I am not updating 

Plus I doubled down when I saw a black Friday deal...

In for a penny!

**Comment by mmornati** [comment]
I don't know if this can help any of you. I had the same problem and I fixed understanding that as I was missing a piece on my side: when I installed (without a proper doc from Sonoff) and because it was working with the previous firmware after the installation, I didn't take care to the little pin delivered with the adapter.
I'm using the Giacomini adapter on my side and I fixed the problem (for the Valve I did not lost it) just adding the little plastic pin which is making the thermostatic valve accessible.
I just need to 3D print all the ones I lost since the installation :(
![PXL_20241218_193804218 MP](https://github.com/user-attachments/assets/931489e3-0baa-4dd4-ab4d-ef195255ee5b)

**Comment by Ofenhed** [comment]
I think all of mine is doing this as well... would be much less of an issue if I could automatically detect it, but I can't find the zigbee specification for the TRVZB, which I know I've read before. Does anyone have an idea of where I could find it, or of a way to monitor the FS error code?

**Comment by CRITCMAR** [comment]
I have 8 trvs that were all working perfectly fine  but have now all faulted to F5. The zigbee bridge is working fine so the issue has to relate to the firmware update!!

**Comment by SirShrimp** [comment]
I ordered two new TRVZBs that had a newer firmware, it seems - same F5 error when installing, older ones (old fw, I guess...) work fine.
I use the RAVL adapter that is supplied without the pin extension.
I designed this little pin extension as a solution that can be 3D printed - works fine for me (tried with the 3rd largest size, worked right away): [https://makerworld.com/en/models/985747#profileId-960360](https://makerworld.com/en/models/985747#profileId-960360)

**Comment by Daniel-zhan-itead** [comment]
We apologize for the inconvenience caused. Our team is working on a new firmware update to address the issue with the TRVZB. To ensure the update resolves the problem without affecting a wider user base, we’re looking for volunteers to participate in a trial test for the new firmware to fix the F5 calibration issue.

If you're willing to work with us, please let me know and send me an email.
daniel.zhan@itead.cc

**Comment by superfrobo** [comment]
Hi Daniel,

It is almost incredible that a simple thermostat valve can have all this problems, the thermostat function is a basic function that my 15 year old students are able to write following my lessons, but ... ok, I appreciate your efforts, and I am available to test the new firmware for you.

Just to let you know, there is no way to update the  SNZB-02D  sensor firmware using the web page, it goes into error, every time.
Best
Maurizio


________________________________

**Comment by Brian911uk** [comment]
found the perfect solution.  If you have recently purchased power sockets for your home they now come with two small plugs which cover the screw holes to give a nice flat finish.  These plugs fit perfectly over actuator to add a few millimetres to its length.  I needed this rather than my normal 5p piece since mine were mounted horizontally and 5p would not stay in place.

They are called screw cap covers but online there are various types and not sure which I had lying around but here is an example.

https://www.ebay.co.uk/itm/274960835584?_ul=GB&rb_itemId=274960835584

**Comment by NedAlex94** [comment]
I also experience this error and I have to say, it has completely sunk my confidence in the product and brand. Thankfully only one updated, so I will just replace it with the old valve.

I think a very welcome initiative from the developers would be an official guide on how to downgrade the firmware version.

**Comment by Solksjaer** [comment]
They have released a new version `4864`.
Let´s hope this issue get´s fixed with this

**Reply by jonathh21** [reply]
> They have released a new version `4864`. Let´s hope this issue get´s fixed with this

Literally just got offered this.

Trying to find some release notes...

**Reply by Solksjaer** [reply]
Exactly, haven´t really tested and wasn´t able to see the release notes. If you find them, please post the link

**Comment by dragosj** [comment]
Updated 10 units and no issues for now.
\Ver is now 1.3.0 from 20241224

**Reply by Solksjaer** [reply]
So far I have only updated one, and it´s been working fine

**Reply by Ofenhed** [reply]
I've updated ~~3 2~~ 3 devices. They are behaving functionally, as far as I can tell, but they are reporting battery differently than before. One went from 30% to 1%, the other went from 36% to 2%. I will update if either device runs out of power in the coming days.

I would assume that the problem is that I run rechargeable batteries, since they have a lower base voltage.

Edit: The update didn't go through on one of the devices. 

Edit 2: I did update 3 devices. I didn't notice that the reason one of the devices didn't update was that it shut off before it could send its new version number and battery state. I've now updated 4, the 4th one got an F4.5(?) error, but seems to work after replacing the batteries.

**Reply by DavidGeorge528** [reply]
I think the firmware update drains quite a lot of battery. All 8 of mine are much lower battery than before, with one dying after updating and requiring new batteries

**Reply by DavidGeorge528** [reply]
Although, the one that died after the update reports as having firmware 1.2.1 and states it doesn't have an update due. So I'm not sure why that is

**Reply by groenmarsmannetje** [reply]
If you do an OTA update for a battery powered device  you better make sure that the battery % is above 70% when you start the update. It will drain the battery like others also mentioned and you really do not want it to stop working in the middle of a firmware update. But thanks for reporting that everything else looks fine for this update.

**Reply by srett** [reply]
> Although, the one that died after the update reports as having firmware 1.2.1 and states it doesn't have an update due. So I'm not sure why that is

I had than once, I think when updating to 1.2.1 or the one before that. The device did update successfully, but still showed the old version number in the UI, and not showing "update available" anymore. So maybe it's just a display glitch, since obviously z2m could tell that no update was possible. And I could tell it was properly updated though since that was the firmware version that added the ability to configure the opening/closing percentage...

**Comment by JanickGers85** [comment]
Hey there, I'm having the F5 issue on one of my valves and I'm out of home, seems that when a valve has this issue you can't upgrade it.
I was wondering if there a way to get the state of the valve to get notified: I don't see nothing on the payload that can be useful for this pourpose,  except from a strange `local_temperature` of 4°C (room temperature is 15°C).

```json
{
    "battery": 64,
    "child_lock": "LOCK",
    "closing_steps": 640,
    "frost_protection_temperature": 7,
    "idle_steps": 79,
    "last_seen": "2025-02-24T17:21:50+01:00",
    "linkquality": 98,
    "local_temperature": 4,
    "local_temperature_calibration": 0,
    "occupied_heating_setpoint": 7,
    "open_window": "ON",
    "running_state": "idle",
    "system_mode": "heat",
    "update": {
        "installed_version": 4609,
        "latest_version": 4864,
        "state": "available"
    },
    "valve_closing_degree": 99,
    "valve_closing_limit_voltage": 2594,
    "valve_motor_running_voltage": 1314,
    "valve_opening_degree": 0,
    "valve_opening_limit_voltage": 1899,
    "weekly_schedule": {
        "friday": "00:00/16 06:00/22.5 20:00/16 20:00/16 20:00/16 20:00/16",
        "monday": "00:00/16 06:00/22.5 20:00/16 20:00/16 20:00/16 20:00/16",
        "saturday": "00:00/16 06:00/22.5 20:00/16 20:00/16 20:00/16 20:00/16",
        "sunday": "00:00/16 06:00/22.5 20:00/16 20:00/16 20:00/16 20:00/16",
        "thursday": "00:00/16 06:00/22.5 20:00/16 20:00/16 20:00/16 20:00/16",
        "tuesday": "00:00/16 06:00/22.5 20:00/16 20:00/16 20:00/16 20:00/16",
        "wednesday": "00:00/16 06:00/22.5 20:00/16 20:00/16 20:00/16 20:00/16"
    }
}
```

**Comment by Solksjaer** [comment]
I´ve finally updated all my devices, and haven´t seen any issues.
As other users mentioned, you should check the battery status before updating.
Updating takes quite long and uses around 25-35% of the battery (depending on the distance to the main coordinator???).
Therefore, ensure that it is at least 50%, to avoid any issues updating.

**Reply by enoversum** [reply]
I've painfully gone through the F5 disaster with two TRVZBs, I exchanged one with a new one after I messed the first one up physically when re-attaching to the radiator. Had a battery state of about 77%, and 58% after the firmware upgrade to 1.3.0. Even the new unit stopped responding and reported F5 after re-inserting the batteries. Through this discussion, I had a hunch and just exchanged batteries with new ones. 
The TRVZB now stopped acting up and opens and closes as I please. If anyone still falls into the F5 rabbit hole, it's good to check with new batteries apparently, even when Zigbee reports they aren't empty.
